src/server/pfs/server/server.go

```diff
@@ -37,3 +38,20 @@ func NewLocalBlockAPIServer(dir string) (pfsclient.BlockAPIServer, error) { // S
 func NewObjBlockAPIServer(dir string, objClient obj.Client) (pfsclient.BlockAPIServer, error) { // SJ: Also bad naming
 	return newObjBlockAPIServer(dir, objClient)
 }
+
+// NewBlockAPIServer creates a BlockAPIServer using the credentials it finds in
+// the environment
+func NewBlockAPIServer(dir string) (pfsclient.BlockAPIServer, error) {
+	var blockAPIServer pfsclient.BlockAPIServer
+	var err error
+	if blockAPIServer, err = newAmazonBlockAPIServer(dir); err == nil {
+		return blockAPIServer, nil
+	}
+	protolion.Errorf("error create Amazon block backend: %s", err.Error())
+	if blockAPIServer, err = newGoogleBlockAPIServer(dir); err == nil {
+		return blockAPIServer, nil
+	}
+	protolion.Errorf("error create Google block backend: %s", err.Error())
+	protolion.Errorf("failed to create obj backend, falling back to local")
+	return NewLocalBlockAPIServer(dir)
+}
```

Comment:
That seems like a legitimate concern to me. We could pass an environment variable to pachd specifying a backend and it would then error if it didn't find the correct secrets for the backend.